File: src/main/java/com/itsci/mju/maebanjumpen/person/service/PersonService.java
package com.itsci.mju.maebanjumpen.person.service;

import com.itsci.mju.maebanjumpen.person.dto.PersonDTO;

import java.util.List;

public interface PersonService {
    List<PersonDTO> getAllPersons();
    PersonDTO getPersonById(int id);
    PersonDTO savePerson(PersonDTO personDto);
    void deletePerson(int id);
    PersonDTO getPersonByUsername(String username);
    PersonDTO updatePerson(int id, PersonDTO personDto);
    PersonDTO updatePersonPictureUrl(int id, String newBaseUrl);
    // เพิ่มเมธอดนี้สำหรับอัปเดต URL ของทุก Person
    void updateAllPersonPictureUrls(String newBaseUrl);

    void updateAccountStatus(int personId, String newStatus); // ⬅️ เพิ่มเมธอดนี้

}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/person/service/PersonService.kt
package com.itsci.mju.maebanjumpen.person.service

import com.itsci.mju.maebanjumpen.person.dto.PersonDTO

interface PersonService {
    fun getAllPersons(): List<PersonDTO>
    fun getPersonById(id: Long): PersonDTO?
    fun savePerson(personDto: PersonDTO): PersonDTO
    fun deletePerson(id: Long)
    fun getPersonByUsername(username: String): PersonDTO?
    fun updatePerson(id: Long, personDto: PersonDTO): PersonDTO
    fun updatePersonPictureUrl(id: Long, newBaseUrl: String): PersonDTO?
    fun updateAllPersonPictureUrls(newBaseUrl: String)
    fun updateAccountStatus(personId: Long, newStatus: String)
}

